ElectronClient/app/app.js

```diff
@@ -121,19 +121,31 @@ class Application extends BaseApplication {
 			case 'NOTE_VISIBLE_PANES_TOGGLE':
 
 				{
-					let panes = state.noteVisiblePanes.slice();
-					if (panes.length === 2) {
-						panes = ['editor'];
-					} else if (panes.indexOf('editor') >= 0) {
-						panes = ['viewer'];
-					} else if (panes.indexOf('viewer') >= 0) {
-						panes = ['editor', 'viewer'];
-					} else {
-						panes = ['editor', 'viewer'];
-					}
+					const getNextLayout = (currentLayout) => {
+						currentLayout = panes.length === 2 ? 'both' : currentLayout[0];
+
+						let paneOptions;
+						if (Setting.value('layout') === Setting.LAYOUT_EDITOR_VIEWER) {
+							paneOptions = ['editor', 'viewer'];
+						} else if (Setting.value('layout') === Setting.LAYOUT_EDITOR_SPLIT) {
+							paneOptions = ['editor', 'both'];
+						} else if (Setting.value('layout') === Setting.LAYOUT_VIEWER_SPLIT) {
+							paneOptions = ['viewer', 'both'];
+						} else {
+							paneOptions = ['editor', 'viewer', 'both'];
+						}
```

Comment:
The reducer should only read and write the state, so you'd want to access `state.settings.layout` instead. Accessing the Setting.XXX constants is fine.